stetho/src/main/java/com/facebook/stetho/inspector/elements/ObjectDescriptor.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class ObjectDescriptor extends Descriptor {
+  private final Map<Class<?>, String> mClassToFriendlyNameCache =
```

Comment:
You probably don't want or need this.  Running indexOf/substring is almost certainly faster than maintaining and looking up in a hash map.  Remember, in order to lookup in a hash map, you must call String.equals on the key which will iterate through each character of the string.  indexOf and substring will iterate through only part of the string.